testng-core/src/main/java/org/testng/TestNG.java

```diff
@@ -420,7 +421,8 @@ public void initializeSuitesAndJarFile() {
         new JarFileUtils(getProcessor(), m_xmlPathInJar, m_testNames, m_parallelMode);
 
     Collection<XmlSuite> allSuites = utils.extractSuitesFrom(jarFile);
-    m_suites.addAll(processCommandLineArgs(allSuites));
+    allSuites.stream().forEach(s -> processParallelModeCommandLineArgs(s));
```

Comment:
it made no sense to check that all testnames exist in alla suite files in the jar. 
and it also made no sense to check processCommandLineArgs a second time to get its side effect. (with suites that was not on the command line even)  

just enabling the parallel+ threadcount if command line argument was given seems like the intended bugfix in:
https://github.com/cbeust/testng/issues/2532

Reply:
@dsankouski  can you have a look at my correction to ensure that we don't break anything for your fix.

Reply:
Seems fine.

Reply:
We don't need to be streaming, if we are going to be using `for..Each`. We can simplify this to `allSuites.forEach(this::processParallelModeCommandLineArgs);`